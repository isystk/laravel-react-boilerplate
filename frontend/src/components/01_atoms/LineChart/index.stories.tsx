import { Meta, Story } from '@storybook/react'
import React from 'react'
import LineChart, { LineChartProps } from './index'

export default {
  title: '01_atoms/LineChart',
  component: LineChart,
} as Meta

const Template: Story = () => {
  const props: LineChartProps = {
    data: [65, 59, 60, 81, 56, 55],
    labels: ['1 月', '2 月', '3 月', '4 月', '5 月', '6 月'],
    title: 'Dataset 1',
    color: 'rgb(255, 99, 132)',
  }
  return <LineChart {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
