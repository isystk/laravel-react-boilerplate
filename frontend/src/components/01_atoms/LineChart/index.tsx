import React, { FC, useEffect, useRef } from 'react'
import { ContainerProps } from 'types'
import { connect } from '@/components/hoc'
import Chart from 'chart.js/auto'

/** LineChartProps Props */
export type LineChartProps = {
  data: Array<number>
  labels: Array<string>
  title: string
  color: string
}
/** Presenter Props */
export type PresenterProps = LineChartProps & {
  canvasRef: HTMLCanvasElement
}

/** Presenter Component */
const LineChartPresenter: FC<PresenterProps> = ({ canvasRef }) => {
  return (
    <>
      <canvas ref={canvasRef}></canvas>
    </>
  )
}

/** Container Component */
const LineChartContainer: React.FC<
  ContainerProps<LineChartProps, PresenterProps>
> = ({ presenter, data, labels, title, color, ...props }) => {
  const canvasRef = useRef<HTMLCanvasElement | null>(null)

  useEffect(() => {
    const lineChart = new Chart(canvasRef.current!, {
      type: 'line',
      data: {
        labels: labels,
        datasets: [
          {
            label: title,
            data: data,
            fill: false,
            borderColor: color,
          },
        ],
      },
    })
    return () => {
      lineChart.destroy()
    }
  }, [data, labels, title, color])

  return presenter({
    canvasRef,
    ...props,
  })
}

export default connect<LineChartProps, PresenterProps>(
  'LineChart',
  LineChartPresenter,
  LineChartContainer
)
