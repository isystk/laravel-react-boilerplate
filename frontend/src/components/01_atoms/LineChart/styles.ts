import { css, keyframes } from '@emotion/css'

export const lineChart = css``
