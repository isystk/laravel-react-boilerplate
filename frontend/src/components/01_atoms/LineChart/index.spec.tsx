import React from 'react'
import { render } from '@testing-library/react'
import LineChart, { LineChartProps } from './index'

describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    const props: LineChartProps = {
      data: [65, 59, 60, 81, 56, 55],
      labels: ['1 月', '2 月', '3 月', '4 月', '5 月', '6 月'],
      title: 'Dataset 1',
      color: 'rgb(255, 99, 132)',
    }
    const { asFragment } = render(<LineChart {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
