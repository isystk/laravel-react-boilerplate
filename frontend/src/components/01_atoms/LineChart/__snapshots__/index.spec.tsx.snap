// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <canvas
    height="0"
    style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
    width="0"
  />
</DocumentFragment>
`;
